### DISCUSSION TITLE
RandRotated

### DISCUSSION THREAD

**Discussion by FabianHeu** [Project-MONAI/MONAI#2139] [Q&A]
I would like to rotate my image label pairs with the RandRotated function to improve robustness: 

RandRotated(keys=["image", "label"], prob=1.0, range_x=1, range_y=1, range_z=1)

But it ends up that it is rotating the image differently to the label, which doesn't make sense and screws up all the labeling. Is this a planned behaviour or am I missing something?

Help appreciated

Fabian

**Comment by FabianHeu** [comment]
Ok found the error in the plotting. It's the same problem than in https://github.com/Project-MONAI/MONAI/discussions/1788
But I dont't really understand why this is not working

**Reply by rijobro** [reply]
If your issue is the same as in #1788, then the effect you're seeing is because each time you call `RandRotated`, it will apply a new random rotation to the data. So if you call it twice, once for `image`, and then again for `label`, then they will have different rotations. If you just call it once with:

```python
transform = RandRotated(keys=["image", "label"], ...)
out_data = transform(in_data)
rotated_im, rotated_label = out_data["image"], out_data["label"]
```

then the same random rotation will have been applied to both the image and the label.

**Reply by rijobro** [reply]
If it still doesn't work, please share a minimum working example.

**Comment by FabianHeu** [comment]
it works now but I still don't get it. This is how it works

```python
training_transform = Compose([
    LoadImaged(keys=['image', 'label']),
    AddChanneld(keys=["image", "label"]),
    RandRotated(keys=["image", "label"], prob=1.0, range_x=1, range_y=1, range_z=1),
    ToTensord(keys=['image', 'label']),`

training_dataset = PersistentDataset(data=training_data_dict, transform=training_transform)

plot_train = training_dataset[0]                                                                                                    # here is the difference
figure1 = plt.figure(num="After_Pipeline", figsize=(8, 4))
plt.suptitle("Image label pair of {}. training-dataset with name: {}".format(i + 1, str3))
plt.subplot(1, 2, 1)
plt.title("image")
plt.xlabel(str1)
saggital_image = plot_train["image"][0, 60, :, :]                                                                        # here is the difference
plt.imshow(saggital_image, cmap="gray")
plt.subplot(1, 2, 2)
plt.title("label")
plt.xlabel(str2)
saggital_label = plot_train["label"][0, 60, :, :]                                                                            # here is the difference
plt.imshow(saggital_label, cmap="gray")
plt.show()
```

An this how it didnt work:

```python
figure1 = plt.figure(num="After_Pipeline", figsize=(8, 4))
plt.suptitle("Image label pair of {}. training-dataset with name: {}".format(i + 1, str3))
plt.subplot(1, 2, 1)
plt.title("image")
plt.xlabel(str1)
saggital_image = training_dataset[0]["image"][0, 60, :, :]                                                   # here is the difference
plt.imshow(saggital_image, cmap="gray")
plt.subplot(1, 2, 2)
plt.title("label")
plt.xlabel(str2)
saggital_label = training_dataset[0]["label"][0, 60, :, :]                                                       # here is the difference
plt.imshow(saggital_label, cmap="gray")
plt.show()
```

**Reply by rijobro** [reply]
Each time you call `training_dataset[0]`, they will be randomly rotated by a different amount. In the top example, you call `training_dataset[0]` once, but in the bottom example you call it twice. Therefore, in the bottom example, `saggital_image` and `saggital_label` have two different rotations applied to them.

**Reply by FabianHeu** [reply]
Thanks a lot! I really appreciate your help. And by the way this framework and your work is really awesome!!
